src/solver/hydro/management/monthly.cpp

```diff
@@ -244,60 +244,55 @@ void HydroManagement::prepareMonthlyOptimalGenerations(double* random_reservoir_
             assert(!Math::Infinite(data.MOL[realmonth]) && "infinite value detected for MOL");
         }
 #endif
-
-        if (study.parameters.hydroDebug)
+        auto writer = study.getWriter();
+        if (study.parameters.hydroDebug && writer)
         {
-            String folder;
-            folder << study.folderOutput << SEP << "debug" << SEP << "solver" << SEP << (1 + y);
-            if (IO::Directory::Create(folder))
+            Yuni::Clob file;
```

Comment:
Does **file** have to be a **Yuni::Clob** ? Or it can be a **std::string** as well ?
About about `String path;` ? **std::string path;** is better